### PR TITLE
feat(analyzer): Add option to skip setup.py analysis of PIP dependencies

### PR DESCRIPTION
**Pull request by haikoschol**
Previously, the PIP package manager has always set the `--analyze-setup-py-insecurely` flag on invocations of `python-inspector`. This is a potential security risk, as `python-inspector` will execute `setup.py` files of resolved dependencies.

It also causes analysis to fail if those `setup.py` files import packages that are not part of the Python standard library (such as `numexpr` in version 2.10.1).

This change makes it configurable whether the flag is set or not.

### CODE REVIEW

File: plugins/package-managers/python/src/main/kotlin/Pip.kt [hunk 1]
```diff
@@ -102,6 +116,8 @@ class Pip(
     ): PythonInspector.Result {
         val operatingSystem = operatingSystemOption ?: OPTION_OPERATING_SYSTEM_DEFAULT
         val pythonVersion = pythonVersionOption ?: detectPythonVersion() ?: OPTION_PYTHON_VERSION_DEFAULT
+        val analyzeSetupPyInsecurely =
+            analyzeSetupPyInsecurelyOption?.toBoolean() ?: OPTION_ANALYZE_SETUP_PY_INSECURELY_DEFAULT
```
Comment (sschuberth) [inline plugins/package-managers/python/src/main/kotlin/Pip.kt:120]: Nit: We prefer to wrap lines as late as possible in this case, so better write

        val analyzeSetupPyInsecurely = analyzeSetupPyInsecurelyOption?.toBoolean()
            ?: OPTION_ANALYZE_SETUP_PY_INSECURELY_DEFAULT

File: plugins/package-managers/python/src/main/kotlin/Pip.kt [hunk 2]
```diff
@@ -87,6 +93,14 @@ class Pip(
         }
     }
 
+    private val analyzeSetupPyInsecurelyOption = options[OPTION_ANALYZE_SETUP_PY_INSECURELY]?.also { analyzeSetupPy ->
+        require(analyzeSetupPy in BOOLEAN_VALUES) {
```
Comment (sschuberth) [inline plugins/package-managers/python/src/main/kotlin/Pip.kt:97]: As it seems a bit odd to have the `BOOLEAN_VALUES` list, I wonder whether we should instead do e.g.

        requireNotNull(analyzeSetupPy.toBooleanStrictOrNull()) {
            "The '$OPTION_ANALYZE_SETUP_PY_INSECURELY' option must be 'true' or 'false', but was '$analyzeSetupPy'."
        }

File: plugins/package-managers/python/src/main/kotlin/Pip.kt (hunk 1 above)
Comment (sschuberth) [reply plugins/package-managers/python/src/main/kotlin/Pip.kt:120]: Also, when using `toBooleanStrictOrNull()` above, this should become `toBooleanStrict()`.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: plugins/package-managers/python/src/main/kotlin/Pip.kt (hunk 2 above)
Comment (haikoschol) [reply plugins/package-managers/python/src/main/kotlin/Pip.kt:97]: That's definitely better. Even better would be if it was possible to use actual booleans in the config file.

Comment (sschuberth) [reply plugins/package-managers/python/src/main/kotlin/Pip.kt]: > if it was possible to use actual booleans in the config file.

True, plugin-specific typed options are not supported, as it's a bit unclear how plugin should declare their expected types...

File: plugins/package-managers/python/src/main/kotlin/Pip.kt [hunk 3]
```diff
@@ -51,6 +53,8 @@ internal val PYTHON_VERSIONS = listOf("2.7", "3.6", "3.7", "3.8", "3.9", "3.10",
  * - *operatingSystem*: The name of the operating system to resolve dependencies for. One of "linux", "macos", or
  *   "windows". Defaults to "linux".
  * - *pythonVersion*: The Python version to resolve dependencies for. Defaults to "3.10".
+ * - *analyzeSetupPyInsecurely*: If "true", `python-inspector` resolves dependencies from setup.py files by executing
+ *   them. This is a potential security risk. Defaults to "true".
```
Comment (sschuberth) [inline plugins/package-managers/python/src/main/kotlin/Pip.kt:57]: Sorry, one more nit, IIRC we aim to document options here in alphabetical order, so this should go first.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
The `PubFunTest` failure is unrelated.

**Pull request merged**